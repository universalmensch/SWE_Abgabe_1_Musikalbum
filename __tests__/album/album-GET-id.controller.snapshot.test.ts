import { afterAll, beforeAll, describe, test } from '@jest/globals';
import axios, { type AxiosInstance, type AxiosResponse } from 'axios';
import {
    host,
    httpsAgent,
    port,
    shutdownServer,
    startServer,
} from '../testserver.js';
import { type AlbumModel } from '../../src/album/rest/album-get.controller.js';
import { HttpStatus } from '@nestjs/common';

const idVorhanden = '1';

describe('GET /rest/:id', () => {
    let client: AxiosInstance;

    beforeAll(async () => {
        await startServer();
        const baseURL = `https://${host}:${port}/rest`;
        client = axios.create({
            baseURL,
            httpsAgent,
            validateStatus: (status) => status < 500, // eslint-disable-line @typescript-eslint/no-magic-numbers
        });
    });

    afterAll(async () => {
        await shutdownServer();
    });

    test('Album zu vorhandener ID', async () => {
        const url = `/${idVorhanden}`;

        const response: AxiosResponse<AlbumModel> = await client.get(url);

        const { status, headers, data } = response;

        expect(status).toBe(HttpStatus.OK);
        expect(headers['content-type']).toMatch(/json/iu);

        // eslint-disable-next-line no-underscore-dangle
        const selfLink = data._links.self.href;

        expect(selfLink).toMatchSnapshot();
    });
});
