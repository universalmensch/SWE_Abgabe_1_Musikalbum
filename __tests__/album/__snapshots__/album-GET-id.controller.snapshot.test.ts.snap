// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /rest/:id Album zu vorhandener ID 1`] = `"https://desktop-tdgithv:3000/rest/1"`;
